python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -396,9 +425,33 @@ def update_config(self, *args, **kwargs):
         return self.__root__.update_config(*args, **kwargs)
     #proxy methods for model
     @property
-    def hy_catchments(self):
-        return self.__root__.hy_catchments
+    def adjustables(self):
+        return self.__root__._catchments
 
     @property
     def strategy(self):
         return self.__root__.strategy
+    
+    def restart(self) -> int:
+        starts = []
+        for catchment in self.adjustables:
+            starts.append(catchment.restart())
+        if all( x == starts[0] for x in starts):
```

Comment:
So, surprisingly if `starts` ends up being empty, this will still evaluate too `True`.

```python
l = []

assert all(x == l[0] for x in l) == True
```

However, then an `IndexError` would be raised when you try to access `starts[0]`. So you need to add a guard or assert of some kind here.

Reply:
Hmm...that kinda depends on the ability of `self.adjustables` to be empty...but it wouldn't be too difficult to simply add
`if starts and all( ... ):`

Reply:
Yeah for sure. I think even if we can guarantee that `self.adjustables` is always non-empty, it's still worth at least a `# SAFETY` comment.

Reply:
addressed in 8222ac14c67ba1a12c4299b498abfd74d1cd2187